### PR TITLE
Pipeline Orchestration task

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[Orchestration Task](https://trello.com/c/wgnZMbVk/340-orchestration-task)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.9-slim-bullseye
```
Comment (saanobhaai) [inline Dockerfile]: Any reason not to pull from `scl3/task_base` (https://github.com/SpeciesConservationLandscapes/task_base/blob/master/Dockerfile)?

File: src/task.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    """
+
+    pipeline = None
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.pipeline = self._get_pipeline(kwargs)
+        self.exit_on_error = kwargs.get("exit_on_error") or False
```
Comment (saanobhaai) [inline src/task.py]: Refactor to use base `raiseonfail` property?
https://github.com/SpeciesConservationLandscapes/task_base/blob/master/task_base/task.py#L36

File: src/task.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            except Exception as err:
+                logging.error(err)
+                has_errors = True
+
+        return not has_errors or not self.exit_on_error
+
+    def calc(self):
+        for task_group in self.pipeline:
+            if self.run_task_group(task_group) is False:
+                raise PipelineError("Task error")
```
Comment (saanobhaai) [inline src/task.py:118]: I might need some reminding on where/why we're raising exceptions...

File: src/task.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+        has_errors = False
+        for future in futures:
+            try:
+                logging.info(future.result())
+            except Exception as err:
+                logging.error(err)
+                has_errors = True
+
+        return not has_errors or not self.exit_on_error
```
Comment (saanobhaai) [inline src/task.py]: Would this be more readable as
`return has_errors and self.exit_on_error`
?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+RUN apt-get update && apt-get install -y apt-transport-https ca-certificates gnupg curl
+RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+RUN apt-get update &&  apt-get install -y google-cloud-sdk
+
+RUN mkdir -p /app
+WORKDIR /app
+COPY ./src/* .
+COPY ./vm/scl-orchestration-vm.jinja .
```
Comment (saanobhaai) [inline Dockerfile]: I've managed to get confused... These files don't exist, but are part of the pipeline-devops. Maybe the Dockerfile and the devops/jinja stuff got moved but not cleaned up here?

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Awesome -- but a couple of questions

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,19 @@
+VM_IMAGE=scl/orchestration-vm
```
Comment (saanobhaai) [inline Makefile]: We should use `scl3` for consistency

File: src/task.py (hunk 4 above)
Comment (gridcell) [reply src/task.py]: I want the `run_group` outcome to be False, showing that it successfully ran, `return has_errors and self.exit_on_error` would say run_group is True if it did not run successfully

**Pull request merged**